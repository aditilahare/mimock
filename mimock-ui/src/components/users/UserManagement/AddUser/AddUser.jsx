import React from 'react';
import { IoPersonAdd } from 'react-icons/io5';
import IconHeader from 'styles/IconHeader';
import { AddUserWrapper, AddUserHeader } from './AddUser.style';
import AddUserForm from './AddUserForm';

export default function AddUser() {
	return (
		<AddUserWrapper data-testid='add-new-user'>
			<AddUserHeader>
				<IconHeader
					dataTestId='add-user-header'
					icon={<IoPersonAdd />}
					title='Add New User'
				/>
			</AddUserHeader>
			<AddUserForm />
		</AddUserWrapper>
	);
}
