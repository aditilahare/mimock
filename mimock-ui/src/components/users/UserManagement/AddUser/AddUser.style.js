import tw from 'tailwind-styled-components';

export const AddUserWrapper = tw.div`
    w-full
    h-full
`;

export const AddUserHeader = tw.div`
    flex
    justify-between
    items-center
    align-middle
    px-4
    py-4
    my-10
    border-b-2
    border-gray-100
    w-11/12
    mx-auto
`;
