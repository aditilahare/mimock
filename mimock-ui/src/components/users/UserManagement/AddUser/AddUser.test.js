import React from 'react';
import { act, render } from '@testing-library/react';
import AddUser from './AddUser';

describe('AddUser', () => {
	it('should render add user component', async () => {
		let tree;
		await act(async () => {
			tree = await render(<AddUser />);
		});

		const { getByTestId, container } = tree;

		expect(getByTestId('add-new-user')).toBeInTheDocument();
		expect(getByTestId('add-user-header')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
