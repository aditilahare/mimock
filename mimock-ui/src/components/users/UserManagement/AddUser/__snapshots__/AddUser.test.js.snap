// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser should render add user component 1`] = `
<div>
  <div
    class="w-full h-full"
    data-testid="add-new-user"
  >
    <div
      class="flex justify-between items-center align-middle px-4 py-4 my-10 border-b-2 border-gray-100 w-11/12 mx-auto"
    >
      <div
        class="flex items-center align-middle font-sans text-2xl font-bold text-gray-700"
        data-testid="add-user-header"
      >
        <button
          aria-label="Go back"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium sc-bdvvtL glRTRM css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          data-testid="go-back-btn"
          tabindex="0"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M401.4 224h-214l83-79.4c11.9-12.5 11.9-32.7 0-45.2s-31.2-12.5-43.2 0L89 233.4c-6 5.8-9 13.7-9 22.4v.4c0 8.7 3 16.6 9 22.4l138.1 134c12 12.5 31.3 12.5 43.2 0 11.9-12.5 11.9-32.7 0-45.2l-83-79.4h214c16.9 0 30.6-14.3 30.6-32 .1-18-13.6-32-30.5-32z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M288 256c52.79 0 99.43-49.71 104-110.82 2.27-30.7-7.36-59.33-27.12-80.6C345.33 43.57 318 32 288 32c-30.24 0-57.59 11.5-77 32.38-19.63 21.11-29.2 49.8-27 80.78C188.49 206.28 235.12 256 288 256zm207.38 183.76c-8.44-46.82-34.79-86.15-76.19-113.75C382.42 301.5 335.83 288 288 288s-94.42 13.5-131.19 38c-41.4 27.6-67.75 66.93-76.19 113.75-1.93 10.73.69 21.34 7.19 29.11A30.94 30.94 0 00112 480h352a30.94 30.94 0 0024.21-11.13c6.48-7.77 9.1-18.38 7.17-29.11zM104 288v-40h40a16 16 0 000-32h-40v-40a16 16 0 00-32 0v40H32a16 16 0 000 32h40v40a16 16 0 0032 0z"
          />
        </svg>
        <div
          class="mx-2 select-none"
        >
          Add New User
        </div>
      </div>
    </div>
    <form
      class="w-11/12 pl-32"
      data-testid="add-user-form"
    >
      <div
        class="flex items-center align-middle my-8"
        data-testid="input-container-name"
      >
        <label
          class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
        >
          Name
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="input-name"
            name="name"
            placeholder="Enter name"
            type="text"
          />
        </div>
      </div>
      <div
        class="flex items-center align-middle my-8"
        data-testid="input-container-userName"
      >
        <label
          class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
        >
          Username
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="input-userName"
            name="userName"
            placeholder="Enter username"
            type="text"
          />
        </div>
      </div>
      <div
        class="flex items-center align-middle my-8"
        data-testid="input-container-password"
      >
        <label
          class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
        >
          Password
        </label>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="input-password"
            name="password"
            placeholder="Enter password"
            type="password"
          />
        </div>
        <div
          class="block h-12 min-h-12 max-h-12"
        >
          <input
            class="block px-3 py-1.5 text-base font-normal text-gray-700 bg-zinc-50 bg-clip-padding border border-solid rounded mx-4 focus:outline-blue-200 border-gray-300"
            data-testid="input-confirmPassword"
            name="confirmPassword"
            placeholder="Confirm password"
            type="password"
          />
        </div>
      </div>
      <div
        class="flex items-center align-middle my-8"
        data-testid="input-container-role"
      >
        <label
          class="w-1/5 font-sans font-bold subpixel-antialiased text-lg text-gray-600"
        >
          User role
        </label>
        <div
          class="w-1/2 flex items-center align-middle mx-4"
        />
      </div>
      <div
        class="flex w-1/2 my-20 flex-row gap-4"
      >
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-lime-700 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-green-300 dark:bg-green-400"
          data-testid="add-user-submit-button"
        >
          Submit
        </button>
        <button
          class="px-9 py-2 shadow-md rounded-md text-center text-base font-semibold cursor-pointer hover:brightness-105 outline-none text-red-800 dark:text-white w-1/4 md:w-1/3 lg:w-1/4 xl:w-1/4 sm:w-1/2 bg-red-300 dark:bg-red-500"
          data-testid="add-user-reset-button"
        >
          Reset
        </button>
      </div>
    </form>
  </div>
</div>
`;
